lib/Connector/XiboAudienceReportingConnector.php

```diff
@@ -293,13 +301,19 @@ public function onRegularMaintenance(MaintenanceRegularEvent $event)
                     // Converting the date into the format expected by the API
                     try {
                         if ($this->timeSeriesStore->getEngine() == 'mongodb') {
-                            $start = $row['start']->toDateTime()->format(DateFormatHelper::getSystemFormat());
-                            $end = $row['end']->toDateTime()->format(DateFormatHelper::getSystemFormat());
+
+                            // Dates are saved in UTC format in MongoDB
+                            // Retrieve the start/end date in the local time zone, rather than UTC format
+                            // Convert the dates to the display time zone if available, otherwise use the CMS time zone
+                            $start = ($displayCache[$displayId]['timeZone'] != null)
+                                ? Carbon::instance($row['start']->toDateTime())->timezone($displayCache[$displayId]['timeZone'])
+                                : Carbon::instance($row['start']->toDateTime())->timezone($defaultTimezone);
+                            $end = ($displayCache[$displayId]['timeZone'] != null)
+                                ? Carbon::instance($row['end']->toDateTime())->timezone($displayCache[$displayId]['timeZone'])
+                                : Carbon::instance($row['end']->toDateTime())->timezone($defaultTimezone);
                         } else {
-                            $start = Carbon::createFromTimestamp($row['start'])
-                                ->format(DateFormatHelper::getSystemFormat());
-                            $end = Carbon::createFromTimestamp($row['end'])
-                                ->format(DateFormatHelper::getSystemFormat());
+                            $start = Carbon::createFromTimestamp($row['start']);
```

Comment:
These dates come from MySQL, which will get created in CMS time zone ✅, however, we still want to convert them into their display time zone, correct?